### ISSUE TITLE
Population.GetDistribution() normalization bug

### ISSUE THREAD

**Issue by LB767** [swharden/ScottPlot#624]
I'm no statistics guy but is this correct behaviour?

I would have expected the distribution curve to be scaled accordingly to the histogram, but it seems to only work if the histogram range of values is 100 exactly.

```cs
var plt = new ScottPlot.Plot(600, 400);

Random rand = new Random(0);
double[] values = DataGen.RandomNormal(rand, pointCount: 10000, mean: 50, stdDev: 5);
hist = new ScottPlot.Statistics.Histogram(values);

double barWidth = hist.binSize * 1.2; // slightly over-side to reduce anti-alias rendering artifacts

plt.PlotBar(hist.bins, hist.countsFrac, barWidth: barWidth, outlineWidth: 0);
plt.PlotScatter(hist.bins, hist.countsFracCurve, markerSize: 0, lineWidth: 2, color: Color.Black);
plt.Title("Normal Random Data");
plt.YLabel("Frequency (fraction)");
plt.XLabel("Value (units)");
plt.Axis(null, null, 0, null);
plt.Grid(lineStyle: LineStyle.Dot);

plt.SaveFig("Advanced_Statistics_Histogram.png");
```

![Advanced_Statistics_Histogram](https://user-images.githubusercontent.com/20526358/99908061-98a24580-2ce0-11eb-977e-ef946fa1ad7a.png)

**Comment by swharden** [comment]
Hi @LB767, thanks for reporting this!

This is indeed an error. It seems that defining a min and max of 0 and 100 makes it work properly, but failing for other limits is certainly a bug.

```cs
var hist = new ScottPlot.Statistics.Histogram(values, min: 0, max: 100);
```

Here's some code based on your example I use to reproduce the problem:

```cs
var plt = new ScottPlot.Plot(600, 400);

Random rand = new Random(0);
double[] values = DataGen.RandomNormal(rand, pointCount: 10000, mean: 50, stdDev: 5);
var hist = new ScottPlot.Statistics.Histogram(values, min: 20, max: 80);

plt.PlotBar(hist.bins, hist.countsFrac, barWidth: hist.binSize * 1.2, outlineWidth: 0);
plt.PlotScatter(hist.bins, hist.countsFracCurve, markerSize: 0, lineWidth: 2, color: Color.Black);

plt.Title("Normal Random Data");
plt.YLabel("Frequency (fraction)");
plt.XLabel("Value (units)");
plt.AxisAuto();
plt.Axis(y1: 0);
plt.Grid(lineStyle: LineStyle.Dot);

plt.SaveFig("test.png");
```

![image](https://user-images.githubusercontent.com/4165489/99911726-579c3800-2cc4-11eb-9b4c-66063465ade3.png)

**Comment by swharden** [comment]
```cs
Console.WriteLine($"countsFrac sum: {hist.countsFrac.Sum()}");
Console.WriteLine($"countsFracCurve sum: {hist.countsFracCurve.Sum()}");
```

```
countsFrac sum: 0.9999999999999998
countsFracCurve sum: 3.3268520264865757  <-- this should be 1
```

I think the error is in the function that calculates the curve, here

https://github.com/swharden/ScottPlot/blob/4bf514cfe65e19d644db25f961a81f30e44caa2d/src/ScottPlot/Statistics/Population.cs#L145-L152

**Comment by swharden** [comment]
@LB767 I figured out a solution and can fix the code in the next release. In the mean time, you can copy/paste this corrected method to create the black curve and begin using it immediately. Thanks again for bringing this to my attention!

```cs
public static double[] GetDistributionCorrected(double[] xs, double stDev, double mean, bool normalize)
{
    double[] ys = new double[xs.Length];
    for (int i = 0; i < xs.Length; i++)
        ys[i] = Math.Exp(-.5 * Math.Pow((xs[i] - mean) / stDev, 2));

    if (normalize)
    {
        double sum = ys.Sum();
        for (int i = 0; i < ys.Length; i++)
            ys[i] /= sum;
    }

    return ys;
}
```

```cs
var plt = new ScottPlot.Plot(600, 400);

Random rand = new Random(0);
double[] values = DataGen.RandomNormal(rand, pointCount: 10000, mean: 50, stdDev: 5);
var hist = new ScottPlot.Statistics.Histogram(values);
plt.PlotBar(hist.bins, hist.countsFrac, barWidth: hist.binSize * 1.2, outlineWidth: 0);

// calculate the fractional curve with the new function
double[] countsFracCurve2 = GetDistributionCorrected(hist.bins, hist.stdev, hist.mean, true);
plt.PlotScatter(hist.bins, countsFracCurve2, markerSize: 0, lineWidth: 2, color: Color.Black);

plt.Title("Normal Random Data");
plt.YLabel("Frequency (fraction)");
plt.XLabel("Value (units)");
plt.AxisAuto();
plt.Axis(y1: 0);
plt.Grid(lineStyle: LineStyle.Dot);

plt.SaveFig("test.png");
```

![image](https://user-images.githubusercontent.com/4165489/99913583-7ac7e700-2cc6-11eb-8c76-44dbdbc080d3.png)

**Comment by LB767** [comment]
Awesome! Looking very much forward to the next major release (multiple Y axes :D)

**Comment by swharden** [comment]
Just fixed this in 4.0.44 (on NuGet now) and it works as expected

```cs
var hist = new ScottPlot.Statistics.Histogram(values);
plt.PlotBar(hist.bins, hist.countsFrac, barWidth: hist.binSize * 1.2, outlineWidth: 0);
plt.PlotScatter(hist.bins, hist.countsFracCurve, markerSize: 0, lineWidth: 2, color: Color.Black);
```
![image](https://user-images.githubusercontent.com/4165489/99921594-63eeb800-2cf9-11eb-944a-0778fbf0692d.png)

Thanks again for noting this issue!
Scott

**Comment by LB767** [comment]
This is seemingly back to how it was with `countsFrac` being normalized but `countsFracCurve` not with Scottplot 4.1.11-beta?
Unless I'm being very stupid I'm getting the same behavior I described in the 1st post, code below updated to 4.1.11

```csharp
var plt = new ScottPlot.Plot(600, 400);

Random rand = new Random(0);
double[] values = DataGen.RandomNormal(rand, pointCount: 1000, mean: 50, stdDev: 20);
var hist = new ScottPlot.Statistics.Histogram(values, min: 0, max: 100);
        
var bar = plt.AddBar(hist.countsFrac, hist.bins);
bar.BarWidth = hist.binSize * 1.2; // oversize to reduce render artifacts
bar.BorderLineWidth = 0;              

plt.AddScatter(hist.bins, hist.countsFracCurve, lineWidth:2, markerSize:0);         

// decorate the plot
plt.XLabel("Value (-)");
plt.YLabel("Fraction (-)");
plt.SetAxisLimits(yMin: 0);
plt.Grid(lineStyle: LineStyle.Dot);

plt.SaveFig("stats_histogram.png");
```

![stats_histogram](https://user-images.githubusercontent.com/20526358/113143355-552a8b80-922c-11eb-97fc-8e7bf3763bae.png)
